src/service.js

```diff
@@ -192,17 +204,30 @@ class Service {
 		};
 
 		this.stopped = () => {
-			if (_.isFunction(this.schema.stopped))
-				return this.Promise.method(this.schema.stopped).call(this);
-
-			if (Array.isArray(this.schema.stopped)) {
-				return this.schema.stopped
-					.reverse()
-					.map(fn => this.Promise.method(fn.bind(this)))
-					.reduce((p, fn) => p.then(fn), this.Promise.resolve());
-			}
+			return new this.Promise((resolve, reject) => {
+				const timeout = setTimeout(reject, this.settings.gracefulStopTimeout || this.broker.options.gracefulStopTimeout);
```

Comment:
Because it is an "internal" service settings, please follow the conventions and use the `$` prefix. Please change it to `this.settings.$gracefulStopTimeout`